### PR TITLE
support upload config files to yarn cluster

### PR DESCRIPTION
**Pull request by riverzzz**
1.suggest spark-submit ... \   
           --files test-import.conf \
              --conf spark.driver.extraClassPath=./  \
              --conf spark.executor.extraClassPath=./  \
              nebula-exchange-2.0.0.jar -c test-import.conf
       or
          spark-submit ... \
                        --files test-import.conf \
                        --conf spark.driver.extraJavaOptions=-Dconfig.file=./test-import.conf \
                        --conf spark.executor.extraJavaOptions=-Dconfig.file=./test-import.conf \
                        nebula-exchange-2.0.0.jar -c test-import.conf
       to dynamic load spark config,the old if use with scheduler will can not use.
2.fix bug fileSystem suggest close itself,if not will java.io.IOException: Filesystem closed
3.suggest decrease log print,if data = 1T,then log print will >1T
4.add ver and edge partition cost

### CODE REVIEW

**Comment by Nicole00** [comment]
Thank you for your contribution! But I have several confuse.
1. Why there are both `--files application.conf` and `-c application.conf` in the submit command, is it necessary to upload the configuration to cluster.
2. If use `--files` to upload configuration to cluster, then how to get it, I didn't see `SparkFiles.get()`  or `System.getenv("SPARK_YARN_STAGING_DIR")`.

**Comment by riverzzz** [comment]
> Thank you for your contribution! But I have several confuse.
> 
> 1. Why there are both `--files application.conf` and `-c application.conf` in the submit command, is it necessary to upload the configuration to cluster.
> 2. If use `--files` to upload configuration to cluster, then how to get it, I didn't see `SparkFiles.get()`  or `System.getenv("SPARK_YARN_STAGING_DIR")`.

1.for example,if you `--files test.conf,test1.conf,test2.conf`, then you can use `-c test1.conf` choose which conf you use,`test1.conf` just a name. 
if you use `--conf spark.driver.extraJavaOptions=-Dconfig.file=./test-import.conf --conf spark.executor.extraJavaOptions=-Dconfig.file=./test-import.conf`,`-c test-import.conf ` is not necessary.
`--files application.conf` will upload file,you not necessary upload. 
2.`--conf spark.driver.extraClassPath=./ --conf spark.executor.extraClassPath=./` will dynamic load conf,you just only choose witch fileName you used.

**Comment by Nicole00** [comment]
> 1.for example,if you `--files test.conf,test1.conf,test2.conf`, then you can use `-c test1.conf` choose which conf you use,`test1.conf` just a name.if you change source code,then you can change to `-c test1` or `-c name`.
> if you use `--conf spark.driver.extraJavaOptions=-Dconfig.file=./test-import.conf --conf spark.executor.extraJavaOptions=-Dconfig.file=./test-import.conf`,`-c test-import.conf ` is not necessary.
> `--files application.conf` will upload file,you not necessary upload.
> 2.`--conf spark.driver.extraClassPath=./ --conf spark.executor.extraClassPath=./` will dynamic load conf,you just only choose witch fileName you used, not required `SparkFiles.get()` or `System.getenv("SPARK_YARN_STAGING_DIR")`.

So what's the purpose or benefit for users to use `--files` to upload config file to cluster. 
And how do you test these changes, my test didn't pass for this pr.  (PS: if you take `-c test1.conf` just as a name but not a file, then you shouldn't have test1.conf file in your path.)

**Comment by riverzzz** [comment]
> > 1.for example,if you `--files test.conf,test1.conf,test2.conf`, then you can use `-c test1.conf` choose which conf you use,`test1.conf` just a name.if you change source code,then you can change to `-c test1` or `-c name`.
> > if you use `--conf spark.driver.extraJavaOptions=-Dconfig.file=./test-import.conf --conf spark.executor.extraJavaOptions=-Dconfig.file=./test-import.conf`,`-c test-import.conf ` is not necessary.
> > `--files application.conf` will upload file,you not necessary upload.
> > 2.`--conf spark.driver.extraClassPath=./ --conf spark.executor.extraClassPath=./` will dynamic load conf,you just only choose witch fileName you used, not required `SparkFiles.get()` or `System.getenv("SPARK_YARN_STAGING_DIR")`.
> 
> So what's the purpose or benefit for users to use `--files` to upload config file to cluster.
> And how do you test these changes, my test didn't pass for this pr. (PS: if you take `-c test1.conf` just as a name but not a file, then you shouldn't have test1.conf file in your path.)

you can look at this `https://stackoverflow.com/questions/53290715/submit-an-application-property-file-with-spark-typesafe-config` first.
There are two situations:
1.submit in local,like the old way,but the principles behind them are different.this is my test.
image
2.use with scheduler,like this,you can upload you config file.this is scheduler location `https://dolphinscheduler.apache.org/en-us/docs/1.3.4/user_doc/system-manual.html`.
image
the two situations I've tested them all, this `--file filename` load config way will adapt to all situations,not only the two situations.

**Comment by Nicole00** [comment]
1. In fact, the uploading operations are optional for users when submitting the application to yarn-cluster. But when submitting the application with local mode or yarn-client mode, `spark.driver.extraClassPath` must not be set. Even the config is set, it does not work. So maybe your client mode test is invalidate.
image

2. Please make sure the test can pass for cluster mode, thank you for your time to do the test. 
   ps: when testing for cluster mode, please make sure there's no config file in your local path, in case the worker that submits application is chosen as driver.

**Comment by riverzzz** [comment]
> 1. In fact, the uploading operations are optional for users when submitting the application to yarn-cluster. But when submitting the application with local mode or yarn-client mode, `spark.driver.extraClassPath` must not be set. Even the config is set, it does not work. So maybe your client mode test is invalidate.
>    image
> 2. Please make sure the test can pass for cluster mode, thank you for your time to do the test.
>    ps: when testing for cluster mode, please make sure there's no config file in your local path, in case the worker that submits application is chosen as driver.

I know in client,this way load config and submit command,user will don't worry about config.
linux local cluster test,
submit with cluster
image
cluster logs
image

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/ErrorHandler.scala [hunk 1]
```diff
@@ -22,7 +22,7 @@ object ErrorHandler {
       }
     } finally {
       errors.close()
-      fileSystem.close()
+//      fileSystem.close()
```
Comment (Nicole00) [inline nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/ErrorHandler.scala:25]: great change！
please check the scalafmt style. Maybe you can remove this line.

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/processor/EdgeProcessor.scala [hunk 2]
```diff
@@ -85,6 +86,7 @@ class EdgeProcessor(data: DataFrame,
         s"${config.errorConfig.errorPath}/${edgeConfig.name}.${TaskContext.getPartitionId}")
       errorBuffer.clear()
     }
+    LOG.info(s"edge part-costTime>>>>>>>>>>>>>>>>>${TaskContext.getPartitionId()}-${System.currentTimeMillis() - startTime}")
```
Comment (Nicole00) [inline nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/processor/EdgeProcessor.scala:89]: Is it better to change `edge part` to `spark partition for edge`? Users may think it's the partition of nebula's space. The real mean is spark's partition.

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/processor/VerticesProcessor.scala [hunk 3]
```diff
@@ -93,6 +94,7 @@ class VerticesProcessor(data: DataFrame,
         s"${config.errorConfig.errorPath}/${tagConfig.name}.${TaskContext.getPartitionId()}")
       errorBuffer.clear()
     }
+    LOG.info(s"ver part-costTime>>>>>>>>>>>>>>>>>${TaskContext.getPartitionId()}-${System.currentTimeMillis() - startTime}")
```
Comment (Nicole00) [inline nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/processor/VerticesProcessor.scala:97]: Ditto, is it better to change `ver part` to `spark partition for vertex`?

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/ErrorHandler.scala (hunk 1 above)
Comment (riverzzz) [reply nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/ErrorHandler.scala]: It has been deleted.

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/processor/VerticesProcessor.scala (hunk 3 above)
Comment (riverzzz) [reply nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/processor/VerticesProcessor.scala]: already change it

**Comment by Nicole00** [comment]
LGTM

**Review by Nicole00 — APPROVED** [review]

**Comment by jamieliu1023** [comment]
@riverzzz Thanks for your contribution! This is Jamie Liu with Nebula Graph. So we would like to send you a Nebula Contributor certificate via email and ship you a mug to mark this special moment! Could you please kindly let us know where we may send the certificate and ship the mug? You can reach me via jamie.liu at vesoft.com. Thanks again!

**Pull request merged**